tests/TestCase/Database/SchemaAwareTypeTest.php

```diff
@@ -0,0 +1,205 @@
+            'Default' => null,
+            'Collation' => null,
+            'Comment' => 'Lorem ipsum',
+        ]);
+
+        $column = $schema->getColumn('field');
+        $this->assertEquals(
+            [
+                'type' => 'schemaawaretype',
+                'length' => null,
+                'null' => true,
+                'default' => null,
+                'comment' => 'Custom schema aware type comment',
+                'precision' => null,
+            ],
+            $column
+        );
+    }
+
+    protected function convertColumnDescriptionPostgresTest()
+    {
+        $dialect = $this->connection->getDriver()->schemaDialect();
+
+        $schema = new TableSchema('table');
+        $dialect->convertColumnDescription($schema, [
+            'name' => 'field',
+            'type' => 'SCHEMAAWARETYPE',
+            'null' => 'YES',
+            'default' => null,
+            'comment' => 'Lorem ipsum',
+            'char_length' => null,
+            'column_precision' => null,
+            'column_scale' => null,
+            'collation_name' => 'en_US.utf8',
+        ]);
+
+        $column = $schema->getColumn('field');
+        $this->assertEquals(
+            [
+                'type' => 'schemaawaretype',
+                'length' => null,
+                'null' => true,
+                'default' => null,
+                'comment' => 'Custom schema aware type comment',
+                'precision' => null,
+            ],
+            $column
+        );
+    }
+
+    protected function convertColumnDescriptionSqliteTest()
+    {
+        $dialect = $this->connection->getDriver()->schemaDialect();
+
+        $schema = new TableSchema('table');
+        $dialect->convertColumnDescription($schema, [
+            'pk' => false,
+            'name' => 'field',
+            'type' => 'SCHEMAAWARETYPE',
+            'notnull' => false,
+            'dflt_value' => 'NULL',
+        ]);
+
+        $column = $schema->getColumn('field');
+        $this->assertEquals(
+            [
+                'type' => 'schemaawaretype',
+                'length' => null,
+                'null' => true,
+                'default' => null,
+                'comment' => 'Custom schema aware type comment',
+                'precision' => null,
+            ],
+            $column
+        );
+    }
+
+    protected function convertColumnDescriptionSqlserverTest()
+    {
+        $dialect = $this->connection->getDriver()->schemaDialect();
+
+        $schema = new TableSchema('table');
+        $dialect->convertColumnDescription($schema, [
+            'name' => 'field',
+            'type' => 'SCHEMAAWARETYPE',
+            'null' => '1',
+            'default' => 'NULL',
+            'char_length' => null,
+            'precision' => null,
+            'scale' => null,
+            'collation_name' => 'Latin1_General_CI_AI',
+        ]);
+
+        $column = $schema->getColumn('field');
+        $this->assertEquals(
```

Comment:
Do you need these tests if you also have the integration tests? Couldn't the comment, length and column type assertions be done for all db vendors in the integration tests as well?

Reply:
Kind of, maybe... problem is that in order to get a simple integration test running a real column type needs to be used. I've opted to use `TEXT` here, as it works for all dialects, that way I could avoid creating all sorts of custom types for the different dialects, but the side effect was that `convertSchemaColumn()` wouldn't be invoked.

Maybe if I remap the `text` type instead of mapping to `schemaawaretype`. I'll give that a try. There would still need to be some vendor specific code though, as only MySQL supports actual comment SQL.

Reply:
> There would still need to be some vendor specific code though, as only MySQL supports actual comment SQL.

I'm not super concerned with preserving schema comments in custom types. That feels like something we could cover in the docs.